install_dev_repos.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+#
+
+"""
+Helper script for installing spyder and external-deps locally in editable mode.
+"""
+
+import argparse
+import os
+import sys
+from logging import Formatter, StreamHandler, getLogger
+from pathlib import Path
+from subprocess import check_output
+
+from importlib_metadata import PackageNotFoundError, distribution
+from packaging.requirements import Requirement
+
+DEVPATH = Path(__file__).resolve().parent
+DEPS_PATH = DEVPATH / 'external-deps'
+BASE_COMMAND = [sys.executable, '-m', 'pip', 'install', '--no-deps']
+
+REPOS = {}
+for p in [DEVPATH] + list(DEPS_PATH.iterdir()):
+    if p.name.startswith('.') or not p.is_dir() and not (
+            (p / 'setup.py').exists() or (p / 'pyproject.toml').exists()):
```

Comment:
```suggestion
    if (
        p.name.startswith('.')
        or not p.is_dir()
        and not ((p / 'setup.py').exists() or (p / 'pyproject.toml').exists()
    ):
```

Formatting improvement.